src/main/java/org/example/jvspringbootfirstbook/service/cart/ShoppingCartService.java

```diff
@@ -0,0 +1,17 @@
+package org.example.jvspringbootfirstbook.service.cart;
+
+import org.example.jvspringbootfirstbook.dto.cart.CartItemDto;
+import org.example.jvspringbootfirstbook.dto.cart.CartItemRequestDto;
+import org.example.jvspringbootfirstbook.dto.cart.CartItemUpdatedDto;
+import org.example.jvspringbootfirstbook.dto.cart.ShoppingCartDto;
+import org.example.jvspringbootfirstbook.model.User;
+
+public interface ShoppingCartService {
+    ShoppingCartDto findUserCart(User user);
+
+    ShoppingCartDto addItem(User user, CartItemRequestDto cartItemRequestDto);
+
+    CartItemDto update(Long itemId, CartItemUpdatedDto updatedDto);
```

Comment:
Since we are only able to update item quantity in this method, let's change this name a bit, for example:

```suggestion
    CartItemDto updateItemQuantity(Long itemId, CartItemUpdatedDto updatedDto);
```